### PR TITLE
Fix GeoJSONSource#loaded sometimes returning true while there are still pending loads (#669)

### PR DESCRIPTION
**Pull request by vanilla-lake**
<changelog>Fix GeoJSONSource#loaded sometimes returning true while there are still pending loads (#669)</changelog>

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Write tests for all new functionality.
 - [x] Manually test the debug page.
 - [x] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

**Comment by HarelM** [comment]
WOW! This is a great and simple solution to this issue!
Is this the only source that is affected from this issue? i.e. do we have other sources that have the same pattern and has this issue as well?

File: src/source/geojson_source.ts [hunk 1]
```diff
@@ -275,16 +274,18 @@ class GeoJSONSource extends Evented implements Source {
             options.data = JSON.stringify(data);
         }
 
+        this._pendingLoads++;
+
         // target {this.type}.loadData rather than literally geojson.loadData,
         // so that other geojson-like source types can easily reuse this
         // implementation
         this.actor.send(`${this.type}.loadData`, options, (err, result) => {
+            this._pendingLoads--;
```
Comment (HarelM) [inline src/source/geojson_source.ts:283]: nit: This seems right but it's not in the place it used to be.
Are you sure this is the right place?

Comment (vanilla-lake) [reply src/source/geojson_source.ts:283]: I'm pretty confident. Had this statement remained after the early return, the counter wouldn't get decremented when the call is abandoned, thus leaving the source in a perpetual "loading" state.

Comment (HarelM) [reply src/source/geojson_source.ts:283]: Cool, thanks!

**Comment by vanilla-lake** [comment]
Thanks @HarelM! 😃 re: other sources potentially having the same issue, I made a quick search for "_loading" and found that `RasterTileSource`, `VectorTileSource`, `ImageSource` and `VideoSource` seem to use similar patterns. Of those, `VectorTileSource` and `ImageSource` are the most likely to exhibit the same issue since they allow setting a different URL.

**Comment by HarelM** [comment]
I guess we can fix them if someone finds it misbehaving later on and not part of this PR...
Can you please update the change log so I can merge?

**Comment by vanilla-lake** [comment]
Sure! Should I add the line under the `main` or `2.0.0` `Bug fixes` section?

**Comment by HarelM** [comment]
2.0

**Comment by vanilla-lake** [comment]
Done!

**Review by HarelM — APPROVED** [review]

**Pull request merged**